splitio/models/telemetry.py

```diff
@@ -25,3 +99,979 @@ def get_latency_bucket_index(micros):
+            },
+            'impressionsQueueSize': int, 'eventsQueueSize': int, 'impressionsMode': string,
+            'impressionsListener': boolean, 'activeFactoryCount': int, 'redundantFactoryCount': int
+        }
+        :type config: dict
+        """
+
+        with self._lock:
+            self._operation_mode = self._get_operation_mode(config[OPERATION_MODE])
+            self._storage_type = self._get_storage_type(config[OPERATION_MODE])
+            self._streaming_enabled = config[STREAMING_ENABLED]
+            self._refresh_rate = self._get_refresh_rates(config)
+            self._url_override = self._get_url_overrides(config)
+            self._impressions_queue_size = config[IMPRESSIONS_QUEUE_SIZE]
+            self._events_queue_size = config[EVENTS_QUEUE_SIZE]
+            self._impressions_mode = self._get_impressions_mode(config[IMPRESSIONS_MODE])
+            self._impression_listener = True if config[IMPRESSIONS_LISTENER] is not None else False
+            self._http_proxy = self._check_if_proxy_detected()
+            self._active_factory_count = config[ACTIVE_FACTORY_COUNT]
+            self._redundant_factory_count = config[REDUNDANT_FACTORY_COUNT]
+
+    def record_ready_time(self, ready_time):
+        """
+        Record ready time.
+
+        :param ready_time: SDK ready time
+        :type ready_time: int
+        """
+        with self._lock:
+            self._time_until_ready = ready_time
+
+    def record_bur_time_out(self):
+        """
+        Record block until ready timeout count
+
+        """
+        with self._lock:
+            self._block_until_ready_timeout = self._block_until_ready_timeout + 1
+
+    def record_not_ready_usage(self):
+        """
+        record non-ready usage count
+
+        """
+        with self._lock:
+            self._not_ready = self._not_ready + 1
+
+    def get_bur_time_outs(self):
+        """
+        Get block until ready timeout.
+
+        :return: block until ready timeouts count
+        :rtype: int
+        """
+        with self._lock:
+            return self._block_until_ready_timeout
+
+    def get_non_ready_usage(self):
+        """
+        Get non-ready usage.
+
+        :return: non-ready usage count
+        :rtype: int
+        """
+        with self._lock:
+            return self._not_ready
+
+    def get_stats(self):
+        """
+        Get config stats.
+
+        :return: dict of all config stats.
+        :rtype: dict
+        """
+        with self._lock:
+            return {
+                BLOCK_UNTIL_READY_TIMEOUT:  self._block_until_ready_timeout,
+                NOT_READY: self._not_ready,
+                TIME_UNTIL_READY: self._time_until_ready,
+                OPERATION_MODE: self._operation_mode,
+                STORAGE_TYPE: self._storage_type,
+                STREAMING_ENABLED: self._streaming_enabled,
+                REFRESH_RATE: {'sp': self._refresh_rate.splits,
```

Comment:
Just a comment that can help to remove some code: 
`TelemetryConfig` can have a reference to the SDK `config` object, and call `self._config[SPLITS_REFRESH_RATE]` instead of `self._refresh_rate.splits`, and the same for other refresh rates, URL overrides, _streaming_enabled, .... This way `_get_refresh_rates` method and `RefreshRates` class can be removed (idem URL related code).

Reply:
That was the original code, Mati wanted to use methods instead of dict